ffi/src/lib.rs

```diff
@@ -545,29 +1061,31 @@ pub unsafe extern "C" fn tx_change_list_entry_at(tx_report_list: *mut TxChangeLi
     Box::into_raw(report)
 }
 
+/// Returns the value at the provided `index` as a `Entid`.
+///
+/// # Panics
+///
+/// If there is no value present at the `index`.
+///
+/// # Safety
+///
+/// Callers must ensure that the pointer to the `TransactionChange` is not dangling .
 #[no_mangle]
 pub unsafe extern "C" fn changelist_entry_at(tx_report: *mut TransactionChange, index: c_int) -> Entid {
     let tx_report = &*tx_report;
     let index = index as usize;
     tx_report.changes[index].clone()
 }
 
-#[no_mangle]
-pub unsafe extern "C" fn store_sync(store: *mut Store, user_uuid: *const c_char, server_uri: *const c_char) -> *mut ExternResult {
-    let store = &mut*store;
-    let user_uuid = c_char_to_string(user_uuid);
-    let server_uri = c_char_to_string(server_uri);
-    let res = store.sync(&server_uri, &user_uuid);
-    Box::into_raw(Box::new(res.into()))
-}
-
+/// destroy function for releasing the memory for `repr(C)` structs.
 #[no_mangle]
 pub unsafe extern "C" fn destroy(obj: *mut c_void) {
```

Comment:
Suppose a consumer calls `destroy(null)`.  We'll silently accept that, where-as `free null` crashes in the real world.  Are we doing consumers a favour?  Something is horribly wrong, and we're not telling them in the way that is most immediately actionable -- with a crash.  Explain why this is the right choice.  (Note that consumers can always wrap `destroy`, if they want to avoid nulls themselves.  And I suppose that consumers can also force a crash if they want it, so perhaps this is moot.  But still, I'd like to understand our rationale.  Is this a convention in Rust FFI implementations?)

Reply:
no convention. I'll revert to crashing on null - we leave it up to the consumer to manage their pointers in every other case so we should do that here too.